src/nfa/nfa.cc

```diff
@@ -312,6 +312,136 @@ struct StackLevel {
     };
 };
 
+struct NodeData {
```

Comment:
What does this struct represent? Some comment would be nice. Also the name should probably make it more explicit that it is connected with computing useful states.